### PR TITLE
DDF-2025 Restore backwards compatibility for ActionProvider

### PR DESCRIPTION
**Pull request by pklinef**
#### What does this PR do?
Restores backwards compatibility for the ActionProvider interface.

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@ricklarsen 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@kcwire
@stustison

#### Any background context you want to provide?
This interface was modified between 2.8.x and 2.9.x in a backwards incompatible way.

#### What are the relevant tickets?
https://codice.atlassian.net/browse/DDF-2025

#### Checklist:
- [x] Documentation Updated
- [ n/a ] Update / Add Unit Tests
- [ n/a ] Update / Add Integration Tests

### CODE REVIEW

**Comment by coyotesqrl** [comment]
👍

**Comment by coyotesqrl** [comment]
There should be a PR to master as well.

**Comment by pklinef** [comment]
I have a branch for master already but I want to avoid creating a second PR until this one is good to go to avoid duplication of effort on two PRs.

**Comment by clockard** [comment]
I didn't see RegistryPublicationActionProvider in the change list. It will need to be updated too.

**Comment by coyotesqrl** [comment]
That class isn't in the 2.9.x branch, but will need to be updated on master.

**Comment by pklinef** [comment]
@coyotesqrl is correct.  The RegistryPublicationActionProvider change will be in the PR for master.

**Comment by stustison** [comment]
👍

**Comment by ricklarsen** [comment]
👍

**Comment by kcwire** [comment]
👍

**Comment by pklinef** [comment]
@stustison @kcwire @coyotesqrl Default methods did not work for some reason with the Aries Blueprint proxy.  Please review the latest changes.

File: platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionProvider.java [hunk 1]
```diff
@@ -13,41 +13,31 @@
...
+     *
+     * @param subject
+     *            object for which the {@link ActionProvider} is requested to provide an
+     *            {@link Action}
      * @return an {@link Action} object. If no action can be taken on the input, then
-     * <code>Collections.emptyList()</code> shall be returned
+     *         <code>null</code> shall be returned
      */
-    <T> List<Action> getActions(T subject);
+    public <T> Action getAction(T subject);
```
Comment (coyotesqrl) [inline platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionProvider.java]: `public` modifier is unnecessary for interfaces.

File: catalog/core/catalog-core-actions/src/main/java/org/codice/ddf/catalog/actions/AbstractMetacardActionProvider.java [hunk 2]
```diff
@@ -74,30 +72,30 @@ protected AbstractMetacardActionProvider(String actionProviderId, String title,
      * {@inheritDoc}
      */
     @Override
-    public <T> List<Action> getActions(T subject) {
+    public <T> Action getAction(T subject) {
```
Comment (rzwiefel) [inline catalog/core/catalog-core-actions/src/main/java/org/codice/ddf/catalog/actions/AbstractMetacardActionProvider.java]: Should we annotate this return value as `@Nullable` ?

**Comment by rzwiefel** [comment]
Should we mark the interfaces that need it with `@Nullable`? since it looks like at least the implementers in some places return null, I'm assuming that we want to allow the interface contract to do so.

**Comment by coyotesqrl** [comment]
Should change the title on the PR and commit since we're not adding default methods.

File: catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java [hunk 3]
```diff
@@ -53,9 +54,8 @@ public DerivedContentActionProvider(ActionProvider actionProvider) {
         if (!canHandle(input)) {
             return Collections.emptyList();
         }
-        // Expect only 1
-        List<Action> resourceActions = resourceActionProvider.getActions(input);
-        if (resourceActions.isEmpty()) {
+        Action resourceActions = resourceActionProvider.getAction(input);
```
Comment (coyotesqrl) [inline catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java]: Make variable name not plural.

File: platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionProvider.java (hunk 1 above)
Comment (pklinef) [inline platform/action/core/platform-action-core-api/src/main/java/ddf/action/ActionProvider.java]: It is but I wanted to return the interface to exactly how it was before.

File: catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java (hunk 3 above)
Comment (pklinef) [inline catalog/core/catalog-core-contentresourcereader/src/main/java/org/codice/ddf/catalog/content/resource/reader/DerivedContentActionProvider.java]: Fixed

**Comment by stustison** [comment]
👍  again

**Comment by coyotesqrl** [comment]
👍

**Pull request merged**